types-core/index.d.ts

```diff
@@ -58,13 +58,15 @@ declare module '@azure/functions-core' {
     function registerHook(hookName: 'postInvocation', callback: PostInvocationCallback): Disposable;
     function registerHook(hookName: 'appStart', callback: AppStartCallback): Disposable;
     function registerHook(hookName: 'appTerminate', callback: AppTerminateCallback): Disposable;
+    function registerHook(hookName: 'log', callback: LogHookCallback): Disposable;
     function registerHook(hookName: string, callback: HookCallback): Disposable;
 
-    type HookCallback = (context: HookContext) => void | Promise<void>;
+    type HookCallback = (context: HookContext) => unknown;
```

Comment:
I am just wondering what necessitated the change to `unknown` here?

Reply:
The log hooks couldn't return `| Promise<void>` like all the other hooks unless I did some serious refactoring, so I changed this to unknown to more easily let the log one just return `void`